app/src/main/java/com/infomaniak/swisstransfer/ui/screen/onboarding/OnboardingScreen.kt

```diff
@@ -0,0 +1,195 @@
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.infomaniak.library.onboarding.OnboardingPage
+import com.infomaniak.library.onboarding.OnboardingScaffold
+import com.infomaniak.swisstransfer.R
+import com.infomaniak.swisstransfer.ui.components.HighlightedText
+import com.infomaniak.swisstransfer.ui.images.AppImages.AppIllus
+import com.infomaniak.swisstransfer.ui.images.ThemedImage
+import com.infomaniak.swisstransfer.ui.images.illus.onboarding.*
+import com.infomaniak.swisstransfer.ui.screen.onboarding.components.AnimatedOnboardingButton
+import com.infomaniak.swisstransfer.ui.theme.Dimens
+import com.infomaniak.swisstransfer.ui.theme.Margin
+import com.infomaniak.swisstransfer.ui.theme.SwissTransferTheme
+import com.infomaniak.swisstransfer.ui.utils.PreviewLargeWindow
+import com.infomaniak.swisstransfer.ui.utils.PreviewSmallWindow
+import kotlinx.coroutines.launch
+
+@Composable
+fun OnboardingScreen(goToMainActivity: () -> Unit) {
+    val pagerState = rememberPagerState(pageCount = { Page.entries.size })
+    val isLastPage by remember { derivedStateOf { pagerState.currentPage >= pagerState.pageCount - 1 } }
+    val coroutineScope = rememberCoroutineScope()
+
+    val isHighlighted = Page.entries.associateWith { rememberSaveable { mutableStateOf(false) } }
+
+    // Start the highlighting of the text when the associated page is reached in the HorizontalPager
+    LaunchedEffect(pagerState.currentPage) {
+        val currentPage = Page.entries[pagerState.currentPage]
+        val setIsHighlighted = isHighlighted[currentPage]?.component2()
+        setIsHighlighted?.invoke(true)
+    }
+
+    BackHandler(pagerState.currentPage > 0) {
+        coroutineScope.launch {
+            pagerState.animateScrollToPage(pagerState.currentPage - 1)
+        }
+    }
+
+    val onboardingPages = buildList {
+        Page.entries.forEach { page ->
+            add(page.toOnboardingPage(isHighlighted))
+        }
+    }
+
+    OnboardingScaffold(
+        pagerState = pagerState,
+        onboardingPages = onboardingPages,
+        bottomContent = {
+            BottomContent(
+                modifier = Modifier
+                    .padding(it)
+                    .consumeWindowInsets(it),
+                isLastPage = { isLastPage },
+                startMainActivity = goToMainActivity,
+                goToNextPage = { coroutineScope.launch { pagerState.animateScrollToPage(pagerState.currentPage + 1) } },
+            )
+        },
+    )
+}
+
+@Composable
+private fun Page.toOnboardingPage(isHighlighted: Map<Page, MutableState<Boolean>>) = OnboardingPage(
+    background = background.image(),
+    illustration = { Illustration(illustration) },
+    text = {
+        TitleAndDescription(
+            page = this,
+            isHighlighted = { isHighlighted[this]?.value ?: false }
+        )
+    }
+)
+
+@Composable
+private fun Illustration(illustration: ThemedImage) {
```

Comment:
This is only used once and is a oneliner
Is is relly interesting to get it in a separate component here ? 
It feels like a unnecessary wrap of the original Image component